src/contrib/ethosu/cascader/common.h

```diff
@@ -103,6 +119,18 @@ inline std::size_t hash_vector(const std::vector<T>& vec) {
   return seed;
 }
 
+template <class T>
+inline T mul_reduce(const std::vector<T>& vec) {
+  if (vec.size() == 0) {
+    return 0;
+  }
+  T v = vec[0];
+  for (unsigned int i = 1; i < vec.size(); i++) {
+    v *= vec[i];
+  }
+  return v;
+}
```

Comment:
Minor suggestion but you could use instead of this template.
```C++
std::accumulate(vec.begin(),vec.end(), 1, std::multiplies<T>());
```

Reply:
Good point, could we take in a follow-up/add on to avoid CI respin?

Reply:
Yes sounds reasonable.

Reply:
I've kept the function as I think it's a useful shorthand, but have changed the implementation to your suggestion.